src/huggingface_hub/hf_api.py

```diff
@@ -1148,6 +1148,100 @@ def list_metrics(self) -> List[MetricInfo]:
         d = r.json()
         return [MetricInfo(**x) for x in d]
 
+    @_deprecate_positional_args
+    def list_spaces(
+        self,
+        *,
+        filter: Union[str, Iterable[str], None] = None,
+        author: Optional[str] = None,
+        search: Optional[str] = None,
+        sort: Union[Literal["lastModified"], str, None] = None,
+        direction: Optional[Literal[-1]] = None,
+        limit: Optional[int] = None,
+        cardData: Optional[bool] = None,
+        datasets: Union[bool, str, Iterable[str], None] = None,
```

Comment:
So indeed this does not work

```py
path = "https://huggingface.co/api/spaces"
params = {"models":True}

r = requests.get(path, params=params)

len(r.json())
``` 
I would keep datasets and models parameters to be just string and list of strings. That way you just use them to specify which datasets/models you want linked. We discussed internally and we think we can remove the boolean